### PR TITLE
Refactor Components, verify-sensei page, widgets, StatusTag

### PR DESCRIPTION
**Pull request by rawrrencee**
# Feature

Refactor Admin page: Verify-sensei page merged into User management page
fix bug when approving/rejecting a course request resets the table
fix variable height of the CountIconWidget when it doesn't act as a button
update StatusTag to support more enum types
add Total Banned, Total Banned Students, Total Banned Senseis skeleton into BannedTable

see screenshots and commit log for more info

## Checklist:

- [ ] Merged latest develop
- [ ] PR title makes sense

## Screenshots (where relevant):
image
image
image
image
image
image

### CODE REVIEW

File: src/components/Admin/UsersManagement/UserCountIconWidget/index.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+          <CountIconWidget
+            title="Rejected Senseis"
+            className={`${currentFilter === 'rejected' ? 'btn btn-light' : 'btn'}`}
+            count={size(rejectedSenseis)}
+            icon={<CloseOutlined />}
+            onClick={() => setTableData('rejected')}
+            color="red"
+          />
+        </div>
+      </div>
```
Comment (natalieloke) [inline src/components/Admin/UsersManagement/UserCountIconWidget/index.js]: mostly similar to the component in ManagementSkeleton? perhaps can consider refactoring this part into a common component, but for next time, given time considerations

Screenshot 2021-03-29 at 10 06 46 AM

File: src/components/Admin/UsersManagement/BannedWidget/index.js [hunk 2]
```diff
@@ -39,11 +39,31 @@ const BannedWidget = () => {
...
   return (
-    <CountIconWidget
-      title="Total Banned Accounts"
-      count={count}
-      icon={<i className="fe fe-slash" />}
-    />
+    <div className="row mt-4">
+      <div className="col-12 col-md-4">
+        <CountIconWidget
+          title="Total Banned Accounts"
```
Comment (y1bs) [inline src/components/Admin/UsersManagement/BannedWidget/index.js]: Not something very important, but I think other pages don't put Total as the first widget right? You wanna standardised to placing them as the last widget?

File: src/components/Common/StatusTag/index.js [hunk 3]
```diff
@@ -1,30 +1,55 @@
...
+        colour = 'geekblue'
+        break
+      case USER_TYPE_ENUM.ADMIN:
+        colour = 'magenta'
+        break
+      default:
+        break
+    }
+    return <Tag color={colour}>{dataSource}</Tag>
   }
```
Comment (natalieloke) [inline src/components/Common/StatusTag/index.js]: moving forward will this be the component that we use for displaying custom tag colours for different data types?

File: src/constants/constants/index.js [hunk 4]
```diff
@@ -5,6 +5,7 @@ export const ADMIN_ROLE_ENUM = {
 }
 
 export const ADMIN_VERIFIED_ENUM = {
+  ENUM_NAME: 'ADMIN_VERIFIED_ENUM',
```
Comment (y1bs) [inline src/constants/constants/index.js]: Actually is ENUM_NAME necessary? Cuz to call it we would have to call ADMIN_VERIFIED_ENUM.ENUM_NAME right? If that's the case than I think no need to use this? Cuz its longer than just typing the name

File: src/constants/constants/index.js [hunk 5]
```diff
@@ -53,6 +54,7 @@ export const STATUS_ENUM = {
 }
 
 export const USER_TYPE_ENUM = {
+  ENUM_NAME: 'USER_TYPE_ENUM',
```
Comment (y1bs) [inline src/constants/constants/index.js]: Same as above

File: src/pages/admin/courses/index.js [hunk 6]
```diff
@@ -144,13 +147,17 @@ const CourseContentManagement = () => {
       dataIndex: 'adminVerified',
       key: 'adminVerified',
       filters: ADMIN_VERIFIED_ENUM_FILTER,
+      width: '10%',
       onFilter: (value, record) => record.adminVerified.indexOf(value) === 0,
-      render: record => <StatusTag data={{ adminVerified: record }} />,
+      render: record => (
+        <StatusTag data={{ adminVerified: record }} type={ADMIN_VERIFIED_ENUM.ENUM_NAME} />
```
Comment (y1bs) [inline src/pages/admin/courses/index.js]: Same as above

File: src/components/Admin/UsersManagement/BannedWidget/index.js (hunk 2 above)
Comment (rawrrencee) [reply src/components/Admin/UsersManagement/BannedWidget/index.js]: okok! will change!

Comment (rawrrencee) [reply src/components/Admin/UsersManagement/BannedWidget/index.js]: btw this one hor I anyhow put one  just as skeleton example, you can change if you want to hahah!

File: src/components/Admin/UsersManagement/UserCountIconWidget/index.js (hunk 1 above)
Comment (rawrrencee) [reply src/components/Admin/UsersManagement/UserCountIconWidget/index.js]: ok i just did this!

File: src/components/Common/StatusTag/index.js (hunk 3 above)
Comment (rawrrencee) [reply src/components/Common/StatusTag/index.js]: i think can use if the situation fits! trying it out to see what your opinions are

File: src/constants/constants/index.js (hunk 4 above)
Comment (rawrrencee) [reply src/constants/constants/index.js]: ok true, ~~i will change back to common text~~

Comment (rawrrencee) [reply src/constants/constants/index.js]: actually the rationale for putting into a declared variable is coz it helps to prevent typo HAHA

Comment (rawrrencee) [reply src/constants/constants/index.js]: i think can keep ? coz there's autocomplete too

Comment (rawrrencee) [reply src/constants/constants/index.js]: @natalieloke maybe you wanna add your thoughts? i'm not sure which is better.. even though typing the declared var is longer but i guess it helps with typos ><

Comment (natalieloke) [reply src/constants/constants/index.js]: lol i understand the typo rationale but also think it's less readable and adds a lot of extra characters

Comment (natalieloke) [reply src/constants/constants/index.js]: maybe a new const file called type?

Comment (natalieloke) [reply src/constants/constants/index.js]:
```
export const TYPE = {
    USER_TYPE_ENUM : 'USER_TYPE_ENUM' 
    ........
} 
```

Comment (natalieloke) [reply src/constants/constants/index.js]: ^ more work with maintaining more const i guess..

Comment (natalieloke) [reply src/constants/constants/index.js]: @rawrrencee u can decide

File: src/components/Admin/UsersManagement/UserCountIconWidget/index.js (hunk 1 above)
Comment (natalieloke) [reply src/components/Admin/UsersManagement/UserCountIconWidget/index.js]: nice thanks

File: src/constants/constants/index.js (hunk 4 above)
Comment (rawrrencee) [reply src/constants/constants/index.js]: ok i think... opt for the string literal? TYPE.USER_TYPE_ENUM doesn't seem to tell us directly what the attr is referring to i feel, and well ya the maintaining of another const a bit shag HAHA

**Review by y1bs — APPROVED** [review]
Lgtm!

**Pull request merged**